packages/tally/src/test-transactions.ts

```diff
@@ -1,25 +1,68 @@
+import { type DelegateActionParams } from '@rabbitholegg/questdk'
+import { Chains } from '@rabbitholegg/questdk-plugin-utils'
+import {
+  type TestParams,
+  createTestCase,
+} from '@rabbitholegg/questdk-plugin-utils'
+
 // https://arbiscan.io/tx/0xb8f32ab6848dce8847782499c1558f5af834f855811746fdbc3ec6fefda0bb24
-export const ARBITRUM_DELEGATION = {
-  blockHash:
-    '0xdf2e9fdcd475d9c28a70c2fd49fd08bb0ed29353ea6b515bfa9c3041f9292031',
-  blockNumber: '135109029',
-  from: '0x3f9e3fbbbe967481222ddaa98e84470d7099381f',
-  gas: '300000',
-  gasPrice: '100000000',
-  maxFeePerGas: '100000000',
-  maxPriorityFeePerGas: '0',
-  hash: '0xb8f32ab6848dce8847782499c1558f5af834f855811746fdbc3ec6fefda0bb24',
-  input:
-    '0x5c19a95c0000000000000000000000003f9e3fbbbe967481222ddaa98e84470d7099381f',
-  nonce: 20,
-  to: '0x912ce59144191c1204e64559fe8253a0e49e6548',
-  transactionIndex: 4,
-  value: '0',
-  type: 'eip1559',
-  accessList: [],
-  chainId: 42161,
-  v: '0',
-  r: '0x1172c91951ebb45b426e5c3c4de13f72432e6c4fa14391f9ea8c2a92e8900292',
-  s: '0x235efaf61057518479145d60232d69be62cc7eeb2661f95432f51bc3642bb767',
-  typeHex: '0x2',
+export const ARBITRUM_TEST: TestParams<DelegateActionParams> = {
+  transaction: {
+    chainId: Chains.ARBITRUM_ONE,
+    from: '0x3f9e3fbbbe967481222ddaa98e84470d7099381f',
+    to: '0x912ce59144191c1204e64559fe8253a0e49e6548',
+    hash: '0xb8f32ab6848dce8847782499c1558f5af834f855811746fdbc3ec6fefda0bb24',
+    input:
+      '0x5c19a95c0000000000000000000000003f9e3fbbbe967481222ddaa98e84470d7099381f',
+    value: '0',
+  },
+  params: {
+    chainId: Chains.ARBITRUM_ONE,
+    project: '0x912ce59144191c1204e64559fe8253a0e49e6548',
+    delegate: '0x3F9e3FBbBE967481222Ddaa98e84470D7099381f',
+  },
 }
+
+// https://optimistic.etherscan.io/tx/0x7294ebf0a19667d9ca028a962112d858e0efaccdd3dc3effc7f4e12493af24b2
+export const OPTIMISM_TEST: TestParams<DelegateActionParams> = {
+  transaction: {
+    chainId: Chains.OPTIMISM,
+    from: '0xa99f898530df1514a566f1a6562d62809e99557d',
+    to: '0x4200000000000000000000000000000000000042',
+    hash: '0x7294ebf0a19667d9ca028a962112d858e0efaccdd3dc3effc7f4e12493af24b2',
+    input:
+      '0x5c19a95c0000000000000000000000003eee61b92c36e97be6319bf9096a1ac3c04a1466',
+    value: '0',
+  },
+  params: {
+    chainId: Chains.OPTIMISM,
+    project: '0x4200000000000000000000000000000000000042',
+    delegate: '0x3EEe61B92C36e97Be6319BF9096A1ac3c04a1466',
+  },
+}
+
+export const passingTestCases = [
```

Comment:
Appriciate that you used the test helpers here 🙏

Reply:
🙏